cranelift-codegen/meta/src/cdsl/formats.rs

```diff
@@ -200,6 +214,14 @@ impl FormatRegistry {
             .expect("unknown InstructionFormat; please define it in shared/formats.rs first")
     }
 
+    pub fn by_name(&self, name: &str) -> InstructionFormatIndex {
+        self.map
+            .iter()
+            .find(|(_key, value)| value.name == name)
+            .unwrap()
```

Comment:
nit: `unwrap_or_else(|| panic!("ahah, I am not the only one to assume that {} is present, but no!", name))`